utils.py

```diff
@@ -759,96 +759,210 @@ def melt_kt_grouped(kt, columns_to_melt, value_column_names, key_column_name='va
     raise NotImplementedError
 
 
-def infer_families(kin_ht,
-                   sex_ht,
-                   i_col='i',
-                   j_col='j',
-                   kin_col='kin',
-                   k2_col='k2',
-                   sex_s_col='s',
-                   sex_sex_col='sex',
-                   first_degree_threshold=(0.23, 0.27),  # TODO: Look at distribution
-                   duplicate_threshold=0.04,  # TODO: Look at distribution
-                   k2_parent_offspring_threshold=0.1  # TODO: Look at distribution
-                   ):
-
-
-    def get_samples_relatives(sample_pairs):
-        samples_rel = defaultdict(set)
-        for row in sample_pairs:
-            samples_rel[row[i_col]].add(row[j_col])
-            samples_rel[row[j_col]].add(row[i_col])
-        return samples_rel
-
-
-    def prune_relatives(relatives):
-        pruned = set()
-        for s, s_rel in relatives:
-            if s not in pruned:
-                pruned.update(s_rel)
-
-        return pruned, {s: s_rel for s, s_rel in relatives.items() if s not in pruned}
-
-
-    def get_fam_samples(s, fam, samples_rel):
-        fam.add(s)
-        for s2 in samples_rel[s]:
+def get_duplicated_samples(
+        kin_ht: hl.Table,
+        i_col: str = 'i',
+        j_col: str = 'j',
+        kin_col: str = 'kin',
+        duplicate_threshold: float = 0.4 # TODO: Look at distribution to pick a value
+) -> Dict[str, Set[str]]: #TODO: Is this the most intuitive format? Otherwise could be List[Set[str]] where each set is a set of duplicated samples.
```

Comment:
I think I like your alternate proposal better. More compact and you can iterate through it to create the one you have now if you really want.